diff --git a/genericadmin/admin.py b/genericadmin/admin.py
index ba72339..345cbee 100644
--- a/genericadmin/admin.py
+++ b/genericadmin/admin.py
@@ -4,7 +4,12 @@
 from django.contrib import admin
 from django.conf.urls import patterns, url
 from django.conf import settings
-from django.contrib.contenttypes import generic
+try:
+    from django.contrib.contenttypes.generic import GenericForeignKey,  GenericTabularInline, GenericStackedInline
+except ImportError:
+    from django.contrib.contenttypes.admin import GenericStackedInline, GenericTabularInline
+    from django.contrib.contenttypes.fields import GenericForeignKey
+
 from django.contrib.contenttypes.models import ContentType
 try:
     from django.utils.encoding import force_text 
@@ -59,7 +64,7 @@ def get_generic_field_list(self, request, prefix=''):
                     field_list.append(fields)
         else:    
             for field in self.model._meta.virtual_fields:
-                if isinstance(field, generic.GenericForeignKey) and \
+                if isinstance(field, GenericForeignKey) and \
                         field.ct_field not in exclude and field.fk_field not in exclude:
                     field_list.append({
                         'ct_field': field.ct_field, 
@@ -143,11 +148,11 @@ class GenericAdminModelAdmin(BaseGenericModelAdmin, admin.ModelAdmin):
     """Model admin for generic relations. """
 
 
-class GenericTabularInline(BaseGenericModelAdmin, generic.GenericTabularInline):
+class GenericTabularInline(BaseGenericModelAdmin, GenericTabularInline):
     """Model admin for generic tabular inlines. """ 
 
 
-class GenericStackedInline(BaseGenericModelAdmin, generic.GenericStackedInline):
+class GenericStackedInline(BaseGenericModelAdmin, GenericStackedInline):
     """Model admin for generic stacked inlines. """
 
 
